File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/consumer/config/ConsumerUniqueGroupIdListener.kt
package io.micronaut.kafka.docs.consumer.config

import io.micronaut.configuration.kafka.annotation.KafkaListener
import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "ConfigProductListenerTest")
// tag::annotation[]
@KafkaListener(groupId = "myGroup", uniqueGroupId = true)
// end::annotation[]
class ConsumerUniqueGroupIdListener {
    // define topic listeners
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/consumer/config/ConsumerUniqueGroupIdListener.py
from micronaut.configuration.kafka.annotation import KafkaListener
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="ConfigProductListenerTest")
# tag::annotation[]
@KafkaListener(groupId="myGroup", uniqueGroupId=True)
# end::annotation[]
class ConsumerUniqueGroupIdListener:
    # define topic listeners
    pass
